docs/index.md

```diff
@@ -0,0 +1,44 @@
+
+<br/>
+
+<!-- https://github.com/bradvin/social-share-urls -->
+[![Facebook](https://img.shields.io/twitter/url/http/shields.io.svg?style=social&label=Share+on+Facebook&logo=facebook)](https://www.facebook.com/sharer/sharer.php?t=small:%20C%2B%2B%20Small%20Containers&u=https://github.com/alandefreitas/small/)
+[![QZone](https://img.shields.io/twitter/url/http/shields.io.svg?style=social&label=Share+on+QZone&logo=qzone)](http://sns.qzone.qq.com/cgi-bin/qzshare/cgi_qzshare_onekey?url=https://github.com/alandefreitas/small/&title=small:%20C%2B%2B%20small%20containers&summary=small:%20C%2B%2B%20small%20containers)
+[![Weibo](https://img.shields.io/twitter/url/http/shields.io.svg?style=social&label=Share+on+Weibo&logo=sina-weibo)](http://sns.qzone.qq.com/cgi-bin/qzshare/cgi_qzshare_onekey?url=https://github.com/alandefreitas/small/&title=small:%20C%2B%2B%20small%20containers&summary=small:%20C%2B%2B%20small%20containers)
+[![Reddit](https://img.shields.io/twitter/url/http/shields.io.svg?style=social&label=Share+on+Reddit&logo=reddit)](http://www.reddit.com/submit?url=https://github.com/alandefreitas/small/&title=Small:%20CPP%20Small%20Containers)
+[![Twitter](https://img.shields.io/twitter/url/http/shields.io.svg?label=Share+on+Twitter&style=social)](https://twitter.com/intent/tweet?text=small:%20C%2B%2B%20small%20containers&url=https://github.com/alandefreitas/small/&hashtags=Containers,Small,Cpp,Programming)
+[![LinkedIn](https://img.shields.io/twitter/url/http/shields.io.svg?style=social&label=Share+on+LinkedIn&logo=linkedin)](https://www.linkedin.com/shareArticle?mini=false&url=https://github.com/alandefreitas/small/&title=small:%20C%2B%2B%20small%20containers)
+[![WhatsApp](https://img.shields.io/twitter/url/http/shields.io.svg?style=social&label=Share+on+WhatsApp&logo=whatsapp)](https://api.whatsapp.com/send?text=small:%20C%2B%2B%20small%20containers:+https://github.com/alandefreitas/small/)
+[![Line.me](https://img.shields.io/twitter/url/http/shields.io.svg?style=social&label=Share+on+Line.me&logo=line)](https://lineit.line.me/share/ui?url=https://github.com/alandefreitas/small/&text=small:%20C%2B%2B%20small%20containers)
+[![Telegram.me](https://img.shields.io/twitter/url/http/shields.io.svg?style=social&label=Share+on+Telegram.me&logo=telegram)](https://telegram.me/share/url?url=https://github.com/alandefreitas/small/&text=small:%20C%2B%2B%20small%20containers)
+[![HackerNews](https://img.shields.io/twitter/url/http/shields.io.svg?style=social&label=Share+on+HackerNews&logo=y-combinator)](https://news.ycombinator.com/submitlink?u=https://github.com/alandefreitas/small/&t=small:%20C%2B%2B%20small%20containers)
+
+<br/>
+
+<!-- START mdsplit-ignore -->
+<!-- Remove link to documentation from the documentation -->
+
+<h2>
+
+[READ THE DOCUMENTATION FOR A QUICK START AND EXAMPLES](https://alandefreitas.github.io/small/)
+
+</h2>
```

Comment:
The pages should end with the macros for the doxygen classes (see `futures`). This allows us to link the library classes in the exposition.

Reply:
Hmmm I'm not quite sure we have this in the futures' index.md.

Reply:
That's correct. That's a comment assuming the pages would be split. 

`index.md` is the only page for which we don't have

https://github.com/alandefreitas/futures/blob/326957b2bc4238716faecd7a5bff4616d967b4ce/docs/motivation.md?plain=1#L634

because it's just the index. It'll probably never have these references.

Reply:
Now I understand. I'll check this out.

Reply:
This is kind of how it works:

- First we have the references generated by doxybook2. 
- Doxygen creates references in XML from the javadoc in the header files. 
- Then doxybook2 (plus the templates in `.doxybook/templates/`) creates a markdown version of the XML in a format similar to cppreference.
- We append these doxygen/doxybook2 documentation files at the end of the exposition ([ref](https://github.com/alandefreitas/futures/blob/326957b2bc4238716faecd7a5bff4616d967b4ce/mkdocs.yml#L30))
- The beginning of the exposition is still manually written, as usual. But the manual exposition can now (and should now) link to elements in the automatically generated doxygen/doxybook2 documentation. 
- To facilitate linking to elements in the automatic doxygen docs, we create a [references.md](https://github.com/alandefreitas/futures/blob/326957b2bc4238716faecd7a5bff4616d967b4ce/docs/references.md?plain=1#L53) file with macros for the links we might need recurrently in the manual exposition. These references are [appended](https://github.com/alandefreitas/futures/blob/326957b2bc4238716faecd7a5bff4616d967b4ce/docs/quickstart.md?plain=1#L218) to all files that might need it.

So, for instance, we can have 

```markdown
[vector]: /small/reference/Classes/classsmall_1_1vector/
```

in the references.md and (assuming this link is correct), we can just use `[vector]` in the exposition every time we would just type `vector`. So whenever the user is reading about `small::vector` in the exposition, we can click it and go straight to the `small::vector` reference in `/small/reference/Classes/classsmall_1_1vector/`. 

Of course, these macros don't have to be limited to our reference, and might include links to the [C++ reference](https://github.com/alandefreitas/futures/blob/326957b2bc4238716faecd7a5bff4616d967b4ce/docs/references.md?plain=1#L4), other [libraries](https://github.com/alandefreitas/futures/blob/326957b2bc4238716faecd7a5bff4616d967b4ce/docs/references.md?plain=1#L35) or whatever references we need recurrently in our exposition. So whenever we find something that might be recurring in the exposition, like class and function names, we include it in references and just use the macro in the exposition instead of having to write something like `[vector](/small/reference/Classes/classsmall_1_1vector/)` over and over again.

Reply:
This is resolved, right?